script-engine/src/main/java/org/ihtsdo/termserver/scripting/reports/qi/TemplateList.java

```diff
@@ -115,10 +117,11 @@ private int compare(Entry<String, List<Template>> entry1, Entry<String, List<Tem
 			if (top1.equals(top2)) {
 				return top1.getFsn().compareTo(top2.getFsn());
 			}
-		} catch (Exception e) {}
-		
-		//In the same major hierarchy or if we have any problems, 
-		return entry1.getKey().compareTo(entry2.getKey());
+		} catch (Exception e) {
+			//ignore
+		}
+
+		return -1;
```

Comment:
This report (once the header issue was fixed) would fail with an IllegalArgumentException with the message "Comparison method violates its general contract". This was thrown because the sorted method (called on line 89) identified that there was a bug within the comparator. I was unable to resolve this so I defaulted to returning -1.